src/contracts/implementations/arbitrable/MultipleArbitrableTransaction.js

```diff
@@ -0,0 +1,333 @@
+
+    const status = transactionArbitrableData.status
+    const timeout = transactionArbitrableData.timeout
+    const lastInteraction = transactionArbitrableData.lastInteraction
+
+    if (status !== contractConstants.STATUS.WAITING_SELLER) {
+      throw new Error(errorConstants.CONTRACT_IS_NOT_WAITING_ON_OTHER_PARTY)
+    } else if (Math.trunc(Date.now() / 1000) <= lastInteraction + timeout) {
+      throw new Error(errorConstants.TIMEOUT_NOT_REACHED)
+    }
+
+    try {
+      return this.contractInstance.timeOutByBuyer(transactionId, {
+        from: account,
+        value: 0
+      })
+    } catch (err) {
+      console.error(err)
+      throw new Error(errorConstants.UNABLE_TO_CALL_TIMEOUT)
+    }
+  }
+
+  /**
+   * Call by seller if buyer is timeout.
+   * @param {string} account - ETH address of user.
+   * @param {number} transactionId - The index of the transaction.
+   * @param {string} contractAddress - ETH address of contract.
+   * @returns {object} The result transaction object.
+   */
+  callTimeOutSeller = async (
+    account = this._Web3Wrapper.getAccount(1),
+    transactionId
+  ) => {
+    await this.loadContract()
+
+    const status = await this.contractInstance.status()
+    const timeout = await this.contractInstance.timeout()
+    const lastInteraction = await this.contractInstance.lastInteraction()
+
+    if (status !== contractConstants.STATUS.WAITING_BUYER) {
+      throw new Error(errorConstants.CONTRACT_IS_NOT_WAITING_ON_OTHER_PARTY)
+    } else if (Date.now() >= lastInteraction + timeout) {
+      throw new Error(errorConstants.TIMEOUT_NOT_REACHED)
+    }
+
+    try {
+      return this.contractInstance.timeOutBySeller(transactionId, {
+        from: account,
+        value: 0
+      })
+    } catch (err) {
+      console.error(err)
+      throw new Error(errorConstants.UNABLE_TO_CALL_TIMEOUT)
+    }
+  }
+
+  /**
+   * Appeal an appealable ruling.
+   * @param {string} account Ethereum account (default account[0]).
+   * @param {number} transactionId - The index of the transaction.
+   * @param {bytes} extraData for the arbitrator appeal procedure.
+   * @param {number} appealCost Amount to pay the arbitrator. (default 0.35 ether).
+   * @returns {object} - The result transaction object.
+   */
+  appeal = async (
+    account = this._Web3Wrapper.getAccount(0),
+    transactionId,
+    extraData = 0x0,
+    appealCost = 0.3
+  ) => {
+    await this.loadContract()
+
+    try {
+      return this.contractInstance.appeal(transactionId, extraData, {
+        from: account,
+        value: this._Web3Wrapper.toWei(appealCost, 'ether')
+      })
+    } catch (err) {
+      console.error(err)
+      throw new Error(errorConstants.UNABLE_TO_RAISE_AN_APPEAL)
+    }
+  }
+
+  /**
+   * Data of the contract
+   * @param {number} transactionId - The index of the transaction.
+   * @returns {object} Object Data of the contract.
+   */
+  getData = async transactionId => {
```

Comment:
This should be called contractId. `transactionId` is a commonly used term for the tx hash. Also you are referencing the contract at an index not a transaction unless I am misunderstanding what this is for

Reply:
I agree but basically I prefer used the wording form the smart contract. And the method is `createTransaction` so the keyword is `transaction` no `contract` (and contract can do reference to a smart contract :p)

Reply:
Can we still do something different than `transactionId` since that is used for tx hashes? Perhaps `transactionIndex`

Reply:
Or `arbitrableTransactionId` would work too and is also more descriptive

Reply:
done